pkg/querier/frontend/frontend_queues.go

```diff
@@ -0,0 +1,86 @@
+package frontend
+
+import (
+	"container/list"
+)
+
+type queueRecord struct {
+	ch     chan *request
+	userID string
+}
+
+type queueManager struct {
```

Comment:
Short comment about this struct would be nice.

Reply:
Added a comment and went with `queueIterator`.  Let me know if you'd like any other changes.

Thanks for the review!